lib/app.rb

```diff
@@ -55,16 +57,11 @@ def run
           end
           @deleted = nil
         when "e"
-          if !@projects || @projects.size == 0
+          if no_projects?
             @output_stream.puts("\e[40;38;5;214mCan't edit any projects\e[0m")
             @output_stream.puts("\e[40;38;5;214mNo projects created\e[0m\n\n")
-            break
-          end
-
-          @output_stream.puts("\e[0;35mEnter a project name:\e[0m")
-          name = @input_stream.gets.chomp
-          if (@current_project = @projects.detect { |p| p.keys.first == name })
-            @output_stream.puts("\e[38;5;40mEditing project: '#{name}'\n\n")
+          elsif get_project_name && switch_project && current_project?
```

Comment:
Is this along the lines of what you were thinking?

Reply:
I think there is a step that would be `get_project_name` && `is_project_name_valid?`. Then switching to the project could happen within the `elsif` body. Not sure if that will work since this is so tricky.